src/thorlabs_cube/driver/tcube/tsc.py

```diff
@@ -0,0 +1,360 @@
+
+    async def get_av_modes(self):
+        """Get the current LED indicator mode bits set by set_av_modes(self, mode_bits)
+
+        :return: The LED mode bits set
+        """
+
+        get_msg = await self.send_request(
+            MGMSG.MOT_REQ_AVMODES, [MGMSG.MOT_GET_AVMODES], _CHANNEL
+        )
+
+        mode_bits = get_msg.data[2:]
+        return mode_bits
+
+    async def set_button_parameters(
+        self, mode, position1, position2, timeout1, timeout2
+    ):
+        """Set button parameters for the front panel buttons.
+
+        :param mode: Mode for the buttons (1 for jog, 2 for position mode).
+        :param position1: Position in encoder counts for the top button.
+        :param position2: Position in encoder counts for the bottom button.
+        :param timeout1: Timeout in ms for position1.
+        :param timeout2: Timeout in ms for position2.
+        """
+
+        payload = st.pack(
+            "<HHllHH", _CHANNEL, mode, position1, position2, timeout1, timeout2
+        )
+
+        await self.send(Message(MGMSG.MOT_SET_BUTTONPARAMS, data=payload))
+
+    async def get_button_parameters(self):
+        """Get the current button parameters.
+
+        :return: A tuple containing (mode, position1, position2, timeout1, timeout2)
+        """
+
+        get_msg = await self.send_request(
+            MGMSG.MOT_REQ_BUTTONPARAMS, [MGMSG.MOT_GET_BUTTONPARAMS], _CHANNEL
+        )
+
+        return st.unpack("<HllHH", get_msg.data[2:])
+
+    async def set_eeprom_parameters(self, msg_id):
+        """Save the current parameters for the specified message in the EEPROM.
+
+        This function sends a request to the device to save the parameter settings
+        of the specified message to the EEPROM memory, ensuring the settings are
+        retained even after the device is powered off.
+
+        :param msg_id: The message ID of the message containing the parameters
+                       that need to be saved in the EEPROM.
+        """
+
+        payload = st.pack("<HH", _CHANNEL, msg_id)
+
+        await self.send(Message(MGMSG.MOT_SET_EEPROMPARAMS, data=payload))
+
+    async def get_status_update(self):
+        """Returns a status update on the specified motor channel
+
+        This function sends a request to the device to return status updates of
+        position, encoder count, status bits on channel one as well as, for future use,
+        returns channel identity two and its associated data which is garbage right now
+
+        :return: integers defining the status bits
+
+        """
+
+        get_msg = await self.send_request(
+            MGMSG.MOT_REQ_STATUSUPDATE, [MGMSG.MOT_GET_STATUSUPDATE], _CHANNEL
+        )
+
+        (
+            position,
+            encoder_count,
+            status_bits,
+            chan_identity_two,
+            _RESERVED,
+            _RESERVED,
+            _RESERVED,
+        ) = st.unpack("<HIIIHIII", get_msg.data[2:])
```

Comment:
Do you mean to be assigning to the `_RESERVED` constant multiple times here?

Reply:
In terms of unpacking, the communication protocol indicates that the last few bytes are reserved for "future use" 
![image](https://github.com/user-attachments/assets/203e4546-9717-4938-b74a-a472f03979b2)
So I was just processing them to still match the expected amount of output bytes, let me know if that's not the design pattern we want to follow.

Reply:
The design pattern is fine. The issue is that this code assigns values to `_RESERVED`, which is a constant that is passed to the device in other messages. If a controller returns a non-zero value in one or more of those bytes, then that value may be assigned to the constant. This is a problem because constant values shouldn't be changed and then we don't know what value gets passed to the controller in other commands. Even worse, because of the multiple assignments, I'm not even sure _which_ value would get assigned to `_RESERVED`. Do tuples deterministically get unpacked in order?

A better implementation would be to assign to an empty variable (`_`). Then the constant value isn't changed and using `_` is idiomatic Python for 'this value isn't used for anything'.

Reply:
So from some research regarding the way we have our classes setup

1. Because `_RESERVED` is a class level instance  from `_Cube`, the linter and I didn't catch the fact that the it's supposed to be `self._RESERVED` in terms of proper referencing.
2. Technically, when `self._RESERVED` would be instantiated in the function, it would be a local instance that would inherit a value of `0x00` given from the constant, however, even if locally a new value is written to `self._RESERVED`, it will not affect it on a global level and therefore, not adjust the constant value that is written to other methods.
3. Python deterministically unpacks [tuples](https://dnmtechs.com/tuple-unpacking-order-in-python-3/)
4. I'm just going to follow the best practice you mentioned because it makes the most sense to me as well, thanks for the feedback